libraries/tools/idl2k/build.gradle

```diff
@@ -5,7 +5,9 @@ buildscript {
 apply plugin: "antlr"
 apply plugin: "kotlin"
 
-configurePublishing(project)
+if (System.getProperty("idl2k.deploy", "false").toBoolean()) {
```

Comment:
`system.idl2k.deploy` teamcity paramater translates to a project property `idl2k.deploy` in Gradle, rather than a JVM system property. So we need to use gradle's `Project.findProperty/getProperty/hasProperty`